pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,18 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
-    String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
-    if (timeout != null) {
+    long result = 0L;
+    if (PGProperty.LOGIN_TIMEOUT.isPresent(props)) {
+      String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = (long) (Float.parseFloat(timeout) * 1000);
       } catch (NumberFormatException e) {
         LOGGER.log(Level.WARNING, "Couldn't parse loginTimeout value: {0}", timeout);
       }
+    } else {
+      result = (long) DriverManager.getLoginTimeout() * 1000;
```

Comment:
Actually thinking about this. If parsing fails we should probably throw an exception as the user probably wanted to set the timeout but somehow failed to set it properly. By not failing we are silently ignoring an error. I don't think that is what users would want.